src/hotspot/os/posix/signals_posix.cpp

```diff
@@ -832,28 +833,28 @@ static void check_signal_handler(int sig) {
   if (os_sigaction == NULL) {
     // only trust the default sigaction, in case it has been interposed
     os_sigaction = (os_sigaction_t)dlsym(RTLD_DEFAULT, "sigaction");
-    if (os_sigaction == NULL) return;
+    if (os_sigaction == NULL) return false;
   }
 
   os_sigaction(sig, (struct sigaction*)NULL, &act);
 
   // Compare both sigaction structures (intelligently; only the members we care about).
-  if (!are_handlers_equal(&act, expected_act)) {
+  // Ignore if the handler is our own crash handler.
+  if (!are_handlers_equal(&act, expected_act) &&
```

Comment:
Pre-existing nit: really this should be called `are_actions_equal`. The poor naming is clearer when you now follow this with  `HANDLER_IS`.

Reply:
Thanks for the suggestion.  Function are_handlers_equal() is renamed to are_actions_equal() in the latest commit.